File: stress/t/flt_norm.c
#include <stdio.h>
#include <math.h>
int main(void){
  long i,n=20; double ss,norm,t,scale=1.5; double a[64];
  for(i=0;i<n;i++){ a[i]=(double)i-5.0; }
  for(i=0;i<n;i++){ a[i]=a[i]*scale; }
  ss=0.0;
  for(i=0;i<n;i++){ t=a[i]*a[i]; ss=ss+t; }
  norm=sqrt(ss);
  printf("norm= %f\n",norm);
  return 0;
}

File: stress/t/flt_norm.f
      PROGRAM FNORM
      IMPLICIT DOUBLE PRECISION (A-H,O-Z)
      INTEGER*8 I,N
      DIMENSION A(0:63)
      N=20
      SCALE=1.5D0
      DO 10 I=0,N-1
        A(I)=DBLE(I)-5.0D0
   10 CONTINUE
      DO 20 I=0,N-1
        A(I)=A(I)*SCALE
   20 CONTINUE
      SS=0.0D0
      DO 30 I=0,N-1
        T=A(I)*A(I)
        SS=SS+T
   30 CONTINUE
      ANORM=SQRT(SS)
      WRITE(*,'(A,F0.6)') 'norm= ',ANORM
      END
